src/mon/OSDMonitor.cc

```diff
@@ -2275,15 +2275,21 @@ class C_AckMarkedDown : public C_MonOp {
     MonOpRequestRef op)
     : C_MonOp(op), osdmon(osdmon) {}
 
-  void _finish(int) override {
-    MOSDMarkMeDown *m = static_cast<MOSDMarkMeDown*>(op->get_req());
-    osdmon->mon->send_reply(
-      op,
-      new MOSDMarkMeDown(
-	m->fsid,
-	m->get_target(),
-	m->get_epoch(),
-	false));   // ACK itself does not request an ack
+  void _finish(int r) override {
+    if (r == 0) {
+      MOSDMarkMeDown *m = static_cast<MOSDMarkMeDown*>(op->get_req());
+      osdmon->mon->send_reply(
+        op,
+        new MOSDMarkMeDown(
+          m->fsid,
+          m->get_target(),
+          m->get_epoch(),
+          false));   // ACK itself does not request an ack
+    } else if (r == -EAGAIN) {
+        osdmon->dispatch(op);
+    } else {
+        ceph_abort_msgf("C_AckMarkedDown: unknown result %d", r);
```

Comment:
```suggestion
        derr << "C_AckMarkedDown: unknown result " << r << dendl;
        ceph_abort();
```

```
/home/jenkins-build/build/workspace/ceph-pull-requests/src/mon/OSDMonitor.cc:1071:31: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
       if (pgs_to_check.size() < g_conf->mon_clean_pg_upmaps_per_chunk * 2) {
           ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/jenkins-build/build/workspace/ceph-pull-requests/src/mon/OSDMonitor.cc: In member function 'virtual void C_AckMarkedDown::_finish(int)':
/home/jenkins-build/build/workspace/ceph-pull-requests/src/mon/OSDMonitor.cc:2291:9: error: 'ceph_abort_msgf' was not declared in this scope
         ceph_abort_msgf("C_AckMarkedDown: unknown result %d", r);
         ^~~~~~~~~~~~~~~
/home/jenkins-build/build/workspace/ceph-pull-requests/src/mon/OSDMonitor.cc:2291:9: note: suggested alternative: 'ceph_abort_msg'
         ceph_abort_msgf("C_AckMarkedDown: unknown result %d", r);
         ^~~~~~~~~~~~~~~
         ceph_abort_msg
make[3]: *** [src/mon/CMakeFiles/mon.dir/OSDMonitor.cc.o] Error 1
```

we don't have `ceph_abort_msgf()` back in mimic.